src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -367,10 +402,40 @@ export default class ConfirmOrderPlugin extends Plugin {
                     adyenCheckoutOptions.selectedPaymentMethodHandler);
     }
 
-    mountPaymentButton(paymentMethodInstance) {
+    mountCustomPayButton(paymentMethodInstance) {
         let confirmButtonContainer = $('<div id="adyen-confirm-button" data-adyen-confirm-button></div>');
         $('#confirmOrderForm').append(confirmButtonContainer);
         paymentMethodInstance.mount(confirmButtonContainer.get(0));
         $('#confirmOrderForm button[type=submit]').remove();
     }
+
+    mountPaymentComponent(paymentMethod, selector, isOneClick = false) {
+        const configuration = Object.assign(paymentMethod, {
+            onSubmit: function(state, component) {
+                this.paymentComponent.find('.loader').show();
+                this.paymentComponent.find('[data-adyen-payment-container]').hide();
+                if (state.isValid) {
+                    let extraParams = {
+                        stateData: JSON.stringify(state.data)
+                    };
+                    let formData = FormSerializeUtil.serialize(this.confirmOrderForm);
+                    this.confirmOrder(formData, extraParams);
+                } else {
+                    component.showValidation();
+                    console.log('Payment failed: ', state);
+                }
+            }.bind(this)
+        });
+        if (!isOneClick && paymentMethod.type === 'scheme') {
+            configuration.enableStoreDetails = true;
```

Comment:
Should this depend on a config or if the storedPaymentMethod PM is enabled?

Reply:
Good point, we could add a config variable for it

Reply:
Also it can be a setting not just for scheme but also for bcmc method